sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/ConnectionProviderSuite.scala

```diff
@@ -68,12 +69,20 @@ class ConnectionProviderSuite
       override def canHandle(driver: Driver, options: Map[String, String]): Boolean = true
       override def getConnection(driver: Driver, options: Map[String, String]): Connection =
         throw new RuntimeException()
+      override def needsModifySecurityConfiguration(
```

Comment:
Should I write the default method to `JdbcConnectionProvider` (return false) and override it only where necessary

Reply:
I'm definitely open to this, but defer to the community on whether that is appropriate or not.

It's worth noting that a default of `false` seems like the right answer to me, but Spark versions 3.1+ implicitly have a default of `true`. 

I think a default of `false` is sensible, but it would change what is currently the default behavior.

Reply:
Bad idea to add default implementation(at least within the trait). Such case a user may think it's not important part of the provider and will work with the default. This is not true for all the secure providers.